test/unit/math/prim/arr/err/is_match_size_test.cpp

```diff
@@ -0,0 +1,35 @@
+#include <stan/math/prim/arr.hpp>
+#include <gtest/gtest.h>
+#include <limits>
+#include <vector>
+
+TEST(ErrorHandling, isMatchingSize) {
+  std::vector<double> a;
+  std::vector<double> b;
+
+  EXPECT_TRUE(stan::math::is_matching_size(a, b));
+
+  a.push_back(3.0);
+  a.push_back(3.0);
+
+  EXPECT_FALSE(stan::math::is_matching_size(a, b));
+
+  b.push_back(3.0);
+  b.push_back(3.0);
+  EXPECT_TRUE(stan::math::is_matching_size(a, b));
+}
+
+TEST(ErrorHandling, isMatchingSize_nan) {
```

Comment:
NaN elements aren't relevant here, but testing for other relevant inputs like 2D arrays, vectors, and matrices are relevant if this is going to be used for that.